addons/test_mail_full/tests/common.py

```diff
@@ -12,6 +12,37 @@ class TestMailFullCommon(TestMassMailCommon, MassSMSCommon):
     def setUpClass(cls):
         super(TestMailFullCommon, cls).setUpClass()
 
+        cls.mailing_sms = cls.env['mailing.mailing'].with_user(cls.user_marketing).create({
+            'name': 'XMas SMS',
+            'subject': 'Xmas SMS for {object.name}',
+            'mailing_model_id': cls.env['ir.model']._get('mail.test.sms').id,
+            'mailing_type': 'sms',
+            'mailing_domain': '%s' % repr([('name', 'ilike', 'MassSMSTest')]),
+            'body_plaintext': 'Dear ${object.display_name} this is a mass SMS with two links http://www.odoo.com/smstest and http://www.odoo.com/smstest/${object.id}',
+            'sms_force_send': True,
+            'sms_allow_unsubscribe': True,
+        })
+
+    @classmethod
+    def _create_mailing_sms_test_records(cls, model='mail.test.sms', partners=None, count=1):
+        """ Helper to create data. Currently simple, to be improved. """
+        Model = cls.env[model]
+        phone_field = 'phone_nbr' if 'phone_nbr' in Model else 'phone'
+        partner_field = 'customer_id' if 'customer_id' in Model else 'partner_id'
+
+        vals_list = []
+        for x in range(0, count):
```

Comment:
Don't think you have to specify the "0".
Could also rename "x" ton "index" or something, I don't like one letter variable.